_posts/2016-03-14-Containers-all-the-way-through.md

```diff
@@ -0,0 +1,95 @@
+
+*   a software abstraction of the hardware (virtual machine)
+*   the OS platform running inside the software abstraction (guest OS)
+*   one or more programs running as processes in the guest OS
+
+Its like running a computer (abstracted as software) inside another computer. And the rest of the fundamentals from the bare metal system applies to this abstraction layer as well. When a process is created inside the virtual system, then the host OS which runs the virtual system might also be spawning one or more processes on the host OS in relation to this process.
+
+<img src="/assets/img/custom/blog/2016-03-14-containers-all-the-way-through/virtual-systems.png" alt="Virtual Systems" title="Virtual Systems" class="img img-center img-responsive style-screengrab">
+
+### Container systems
+
+Now looking at container systems we can say the following:
+
+*   they run on top of OS platforms running inside bare metal systems or virtual systems
+*   containers which allow to isolate processes and share the kernel between each other (such isolation from other processes and resources are possible in some OSes like say Linux, due to OS kernel features like _cgroups_ and _namespaces_)
+
+A container creates an OS like environment, inside which one or more programs can be executed. Each of these executions could result in a one or more processes on the host OS. Container systems are composed of these layers:
+
+*   hardware (shared)
+*   the OS platform (shared kernel)
+*   one or more programs running inside the container
+
+<img src="/assets/img/custom/blog/2016-03-14-containers-all-the-way-through/container-systems.png" alt="Container Systems" title="Container Systems" class="img img-center img-responsive style-screengrab">
+
+### Summary
+
+Looking at these enclosures or rounded rectangles within each other, we can already see how its containers all the way through.
+
+There is an increasing number of distinctions between **Bare Metal Systems**, **Virtual Systems** and **Container Systems**. While Virtual Systems encapsulate the operating system inside a thick hardware virtualisation, Container systems do something similar but with a much thinner virtualisation layer.
+
+There are a number of pros and cons between these systems when we look at them individually, i.e. portability, performance, resource consumption, time to recreate such systems, maintaining these systems, et al.
+
+### Word of thanks and stay in touch
+
+Thank you for your time, feel free to send your queries and comments to [theNeomatrix369](http://twitter.com/theNeomatrix369). Big thanks to my colleague, and our DevOps craftsman [Robert Firek](https://twitter.com/robertfirek) from [Codurance](http://codurance.com/aboutus/ourcompany/) for proof-reading my post and steering me in the right direction.
+
+### Resources
+
+* [1] [Hypervisor](https://en.wikipedia.org/wiki/Hypervisor)
```

Comment:
Maybe add the source of the links in the text? Example: "Wikipedia page for hypervisor"

Same for the StackOverflow page